src/agent/include/register.hpp

```diff
@@ -1,8 +1,16 @@
 #pragma once
 
+#include <http_client.hpp>
+
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <functional>
 #include <optional>
 #include <string>
 
+namespace beast = boost::beast;
```

Comment:
Beast doesn't seem necessary, but I think that it is preferable to not introduce these namespace declarations on header files, any other file including this header will "inherit" them, polluting the global namespace.

Reply:
Fixed!